analyses/molecular-subtyping-pathology/02-incorporate-clinical-feedback.Rmd

```diff
@@ -0,0 +1,200 @@
+                                       "Kids_First_Participant_ID")) 
+
+```
+
+#### Output file
+
+```{r}
+output_file <- file.path(results_dir, "compiled_molecular_subtypes_with_clinical_feedback.tsv")
+```
+
+
+Identify pnoc003 samples that have differences between clinical review subtype
+and molecular subtype from histone variant mutation status
+
+```{r}
+diff_subtype <- subtype %>% left_join(pnoc003_review,by=c("Kids_First_Participant_ID","tumor_descriptor")) %>% 
+  dplyr::filter(
+    # checking for samples where clinically reviewed subtype is H3 WT
+    # but molecular subtype is not HGG
+    (grepl("H3 WT",.$`H3 Status`) & !grepl("HGG",molecular_subtype)) |
+    # checking for samples where clinically reviewed subtype is H3 K28M
+    # but molecular subtype is not DMG H3 K28
+    (grepl("K28M",.$`H3 Status`) & !grepl("DMG, H3 K28",molecular_subtype)) | 
+    # checking for samples were clinical review was not found  
+    grepl("no report",.$`H3 Status`) )  %>%
+  dplyr::select(Kids_First_Participant_ID,
+                sample_id,
+                `H3 Status`,
+                molecular_subtype,
+                tumor_descriptor)
+
+diff_subtype
+
+```
+
+Counts in each type of differences to identify how many samples already 
+have subtyping updated from molecular-subtype-HGG module and how many
+samples need molecular subtype updated from clinical review
+
+```{r}
+
+diff_subtype %>% 
+  group_by(`H3 Status`,molecular_subtype) %>%
+  tally()
+
+```
+
+
+Seems like 6 sample (WXS and matching RNA-Seq from PT_NK8A49X5, PT_QA9WJ679, PT_WGVEF96B) molecular subtype 
+need to be updated and Notes columns for these samples will be updated to capture this 
+information
+
+```{r}
+subtype_clinical_review_df <- subtype %>%
+  # matching only on "Kids_First_Participant_ID","tumor_descriptor"
+  left_join(pnoc003_review,by=c("Kids_First_Participant_ID","tumor_descriptor")) %>% 
+  dplyr::mutate(
+    Notes =  case_when(
+      (grepl("H3 WT",.$`H3 Status`) & !grepl("HGG",molecular_subtype)) 
+      ~ paste(Notes,"Updated to HGG, H3 wildtype from clinical review",sep=","),
+      (grepl("K28M",.$`H3 Status`) & !grepl("DMG, H3 K28",molecular_subtype)) 
+      ~ paste(Notes,"Updated to DMG, H3 K28 from clinical review", sep=","),
+      TRUE ~ Notes
+    ),
+    molecular_subtype =  case_when(
+      (grepl("H3 WT",.$`H3 Status`) & grepl("HGG",molecular_subtype)) |
+        (grepl("K28M",.$`H3 Status`) & grepl("DMG, H3 K28",molecular_subtype)) 
+      ~ molecular_subtype,
+      # Don't think we would need to ever change from K28 to H3 wildtype
+      # since there would be evidence for K28 from snv data for the sample
+      #(grepl("H3 WT",.$`H3 Status`) & !grepl("HGG",molecular_subtype)) ~ 
+      #  gsub("DMG, H3 K28","HGG, H3 wildtype",molecular_subtype),
+      (grepl("K28M",.$`H3 Status`) & !grepl("DMG, H3 K28",molecular_subtype)) ~ 
+        gsub("HGG, H3 wildtype","DMG, H3 K28",molecular_subtype),
```

Comment:
As I'm looking at this, I am wondering if, for the purposes of this notebook, we might want to standardize the labels between the `H3 Status` and `molecular_subtype` so you can essentially ask "are these the same?" That logic might be more straightforward. Thoughts? We'd probably also want to filter out the no reports - I think this is essence what this is accomplishing as well.

Reply:
I suppose this extends to the `Notes` step, too.

Reply:
I thought about changing the values in H3 Status to what we have in molecular_subtype, but since we have some additional values  from IDH/BRAF alterations I assumed that it would be the same logic of `grepl` because of the IDH/BRAF alterations subtyping that is not part of clinical review. For example: 

Kids_First_Participant_ID | sample_id | Kids_First_Biospecimen_ID | molecular_subtype | integrated_diagnosis | short_histology | broad_histology | Notes | tumor_descriptor
-- | -- | -- | -- | -- | -- | -- | -- | --
PT_0MXPTTM3 | 7316-3220 | BS_8SYN7GXG | HGG, IDH | High-grade glioma | HGAT | Diffuse astrocytic and oligodendroglial tumor | Updated via OpenPBTA subtyping | Initial CNS Tumor
  | PT_0MXPTTM3 | A09410 | BS_R7NTZR4C | HGG, IDH | High-grade glioma | HGAT | Diffuse astrocytic and oligodendroglial tumor | Updated via OpenPBTA subtyping | Initial CNS Tumor
  | PT_0MXPTTM3 | A09410 | BS_H3TYF6GD | HGG, IDH | High-grade glioma | HGAT | Diffuse astrocytic and oligodendroglial tumor | Updated via OpenPBTA subtyping | Initial CNS Tumor
PT_HGM20MW7 | 7316-3221 | BS_FK3B5SDH | DMG, H3 K28, BRAF V600E | Diffuse midline glioma | HGAT | Diffuse astrocytic and oligodendroglial tumor | Updated via OpenPBTA subtyping | Initial CNS Tumor
  | PT_HGM20MW7 | A09446 | BS_5VPM0F36 | DMG, H3 K28, BRAF V600E | Diffuse midline glioma | HGAT | Diffuse astrocytic and oligodendroglial tumor | Updated via OpenPBTA subtyping | Initial CNS Tumor
  | PT_HGM20MW7 | A09446 | BS_52ETE050 | DMG, H3 K28, BRAF V600E | Diffuse midline glioma | HGAT | Diffuse astrocytic and oligodendroglial tumor | Updated via OpenPBTA subtyping | Initial CNS Tumor


Maybe these IDH, BRAF alterations should also be clinical reviewed so we can match the values?

Reply:
I see - so in the `PT_0MXPTTM3` case, H3 status is wildtype which agrees with the molecular subtyping status and in the `PT_HGM20MW7` H3 status and the subtyping is in agreement, it's just the matter of BRAF V600E also being included.

Taking a step back - my understanding of this addition is that we're essentially looking for any K28M mutations that were _missed_ because they were not detected in the SNV consensus for whatever reason (from #751 and some additional context in #735). I had some ideas about how we might approach this differently for clarity reasons, but I think we can go with this solution and have you add comments throughout at each condition within calls to `case_when()` (lines 180-197).